ext/SpeedyWeatherGeoMakieExt.jl

```diff
@@ -121,4 +122,180 @@ function SpeedyWeather.globe(
     return fig
 end
 
-end # module
\ No newline at end of file
+
+"""
+$(TYPEDSIGNATURES)
+Create an animation from a SpeedyWeather simulation NetCDF output file. Needs a backend like 
+CairoMakie or GtkMakie to be loaded at the time of calling this function.
+
+# Arguments
+- `nc_file::String`: Path to the NetCDF file containing SpeedyWeather simulation data
+- `variable::String = "temp"`: Variable to animate (e.g., "temp", "pres", "vor", "u", "v", "humid")
+- `level::Int = 1`: Vertical level to plot (for 3D variables)
+- `transient_timesteps::Int = 0`: Number of timesteps to skip at the beginning of the animation
+- `output_file::String = "animation.mp4"`: Path to save the animation
+- `colormap = :viridis`: Colormap to use for the animation
+- `framerate::Int = 15`: Frame rate of the animation
+- `title::String = ""`: Title for the animation (if empty, will use variable name)
+- `colorrange = nothing`: Range for the colorbar (nothing for automatic)
+- `projection = "+proj=robin"`: Map projection to use (e.g., "+proj=robin", "+proj=ortho")
+- `figure_size = (800, 600)`: Size of the figure
+
+# Example
+```julia
+using SpeedyWeather, GeoMakie, CairoMakie
+
+# Animate temperature at level 1
+animate_simulation("output.nc", variable="temp", level=1)
+
+# Animate surface pressure with a specific colormap
+animate_simulation("output.nc", variable="pres", colormap=:thermal)
+
+# Create an animation with Orthographic projection
+animate_simulation("output.nc", variable="temp", projection="+proj=ortho +lon_0=0 +lat_0=30")
+```
+"""
+function SpeedyWeather.animate_simulation(
+    nc_file::String;
```

Comment:
Not necessarily for this PR but should we have a lower-level version of this function taking in a 3D ring grid which animates the last dimension? A netCDF array could then be wrapped again into that.

Reply:
But for here, shouldn't we allow for a NCDataset as input rather than a string? Just wondering whether there's an advantage of doing the open and close internally?

Reply:
> Not necessarily for this PR but should we have a lower-level version of this function taking in a 3D ring grid which animates the last dimension? A netCDF array could then be wrapped again into that.

On the other hand, with a netCDF dataset you don't need to load the whole dataset into RAM like this. I can see the point though, maybe something for another PR.

Reply:
> But for here, shouldn't we allow for a NCDataset as input rather than a string? Just wondering whether there's an advantage of doing the open and close internally?

The advantage is not having to open and close it when you just want a quick animation. But I'll do a the low level version with a NCDataset as an input and then have a convinience version with a string input and Simulation input.